src/libraries/System.Data.Common/src/System/Data/SQLTypes/SQLGuid.cs

```diff
@@ -95,18 +96,17 @@ public Guid Value
 
         public byte[]? ToByteArray()
         {
-            byte[] ret = new byte[SizeOfGuid];
-            m_value!.CopyTo(ret, 0); // TODO: NRE
-            return ret;
+            if (_value is null)
```

Comment:
This is an observable behavioral difference, as it takes something that previously threw NRE and changes it to return null.

If you wanted to keep the original NRE behavior, something like this would work:

```cs
byte[]? ret = _value?.ToByteArray();
if (ret!.GetType() == typeof(byte[])) { /* GetType forces null check */ }
return ret;
```

That second line is heavily optimized by the JIT to be a single "throw a NRE if this value is null" instruction.

Reply:
Since the previous behaviour seemed to be a bug (at which the `TODO` and the fact that the method has a nullable return type hint at), I'd like the code owners to decide on this.

Reply:
I don't see any usages on the SqlClient side that depend on the NRE behavior or would bubble that up anywhere. So I'm fine with the change.